src/Generator/Formatting.fs

```diff
@@ -0,0 +1,431 @@
+﻿module Formatting
+open Types
+open Constants
+open TypeMapping
+
+type WriteStatement =
+    | WriteLine of string
+    | Write of string
+    | Indent
+    | Unindent
+
+module WriteStatement =
+    type Configuration =
+        { IndentSize : int }
+        static member Default = { IndentSize = 1 }
+    let writeLine s = WriteLine s
+    let write s = Write s
+    let indent = Indent
+    let unindent = Unindent
+
+    let executeWith conf statements =
+        use backing = new System.IO.StringWriter()
+        use writer = new System.CodeDom.Compiler.IndentedTextWriter(backing)
+        let executeStatement statement =
+            match statement with
+            | WriteLine str -> writer.WriteLine str
+            | Write str -> writer.Write str
+            | Indent -> writer.Indent <- writer.Indent + 1
+            | Unindent -> writer.Indent <- max 0 (writer.Indent - 1)
+        statements
+        |> Seq.iter executeStatement 
+        writer.Flush()
```

Comment:
do you actually need to flush the memory since use will clear out the memory when exiting the scope of your function?

Reply:
I just want to ensure everything has been written to the writer. Maybe this is completely unnecessary, will have a look